doc/settings.md
# Configuration options

## ADC settings

### ADCREF

Source of the ADC reference voltage. Make sure that `ADCUREF` value is correct.

Type     | Size
-------- | --------
enum     | 1 byte 

Value | Source
----- | --------------
 0    | AREF          
 1    | AVCC          
 3    | Internal 1.1V 

### ADCUREF

ADC reference voltage, volts.

Type     | Size
-------- | --------
float    | 4 bytes


## MAX7456 settings

### VMODE

OSD composite video mode. Used only when jumper "PAL" on MinimOSD is not closed.
Value will be applied after OSD reboot.

Type     | Size
-------- | --------
enum     | 1 byte 

Value | Video mode
----- | ----------
 0    | PAL
 1    | NTSC

### VBRIGHT

OSD character brightness in % of OSD white level.
Value will be applied after OSD reboot.

Type     | Size
-------- | --------
enum     | 1 byte 

Value | White | Black 
----- | ----- | -----
 0    | 120%  | 0%
 1    | 100%  | 0%
 2    | 90%   | 0%
 3    | 80%   | 0%
 4    | 120%  | 10%
 5    | 100%  | 10%
 6    | 90%   | 10%
 7    | 80%   | 10%
 8    | 120%  | 20%
 9    | 100%  | 20%
 10   | 90%   | 20%
 11   | 80%   | 20%
 12   | 120%  | 30%
 13   | 100%  | 30%
 14   | 90%   | 30%
 15   | 80%   | 30%


## OSD settings

### SCREENS

Usable screens count, 1..8

Type     | Size
-------- | --------
uint8    | 1 byte 

### SWM

Screen switch mode.

- **Selector** - Range of the input channel (`SELMAX` - `SELMIN`) divided by screens count `SCREENS`.
  Current screen selected by input channel value.
- **Toggle** - Cycle through the screens when the input channel value is crossing threshold `TGTRESH`.

Type     | Size
-------- | --------
enum     | 1 byte 

Value | Mode
----- | -----------------
 0    | Selector mode
 1    | Toggle mode

### SWCH

Input channel used for screens switching.

Type     | Size
-------- | --------
uint8    | 1 byte 

### SELMIN

Minimal pulse width of the input channel (usually 1000). Used in **Selector** mode.

Type     | Size
-------- | --------
uint16   | 2 bytes

### SELMAX

Maximal pulse width of the input channel (usually 2000). Used in **Selector** mode.

Type     | Size
-------- | --------
uint16   | 2 bytes

### TGTRESH

Screen switch threshold, microseconds. Used in **Toggle** mode. Recommended value is in range **1200 .. 1600**.

Type     | Size
-------- | --------
uint16   | 2 bytes


## Telemetry settings

### MINCV

Minimal cell voltage (dead battery), volts. Typical value for LiPo battery is **3.2**.

Type     | Size
-------- | --------
float    | 4 bytes

### NOMCV

Nominal cell voltage, volts. Typical value for LiPo battery is **3.7**.

Type     | Size
-------- | --------
float    | 4 bytes

### MAXCV

Maximal cell voltage, volts. Typical value for LiPo battery is **4.2**.

Type     | Size
-------- | --------
float    | 4 bytes

### LOWCV

Warning threshold, volts. Typical value for LiPo battery is about **3.5**.

Type     | Size
-------- | --------
float    | 4 bytes

### RLT

RSSI low threshold, %

Type     | Size
-------- | --------
uint8    | 1 byte 

### CSIGN

Callsign text, 8 chars max.

Type     | Size
-------- | --------
str      | 9 bytes


## ADCBattery module settings

### ABINT

Update interval, ms.

Type     | Size
-------- | --------
uint16   | 2 bytes

### ABU1CH

ATmega328p ADC channel for voltage sensor of the flight battery.

Type     | Size
-------- | --------
enum     | 1 byte 

Value | ADC Channel | ATMega pin
----- | ----------- | ----------
 0    | 0           | 23
 1    | 1           | 24
 2    | 2           | 25
 3    | 3           | 26
 4    | 4           | 27
 5    | 5           | 28
 6    | 6           | 19
 7    | 7           | 22

### ABU2CH

ATmega328p ADC channel for voltage sensor of the secondary battery.

Type     | Size
-------- | --------
enum     | 1 byte 

Value | ADC Channel | ATMega pin
----- | ----------- | ----------
 0    | 0           | 23
 1    | 1           | 24
 2    | 2           | 25
 3    | 3           | 26
 4    | 4           | 27
 5    | 5           | 28
 6    | 6           | 19
 7    | 7           | 22

### ABI1CH

ATmega328p ADC channel for current sensor of the flight battery.

Type     | Size
-------- | --------
enum     | 1 byte 

Value | ADC Channel | ATMega pin
----- | ----------- | ----------
 0    | 0           | 23
 1    | 1           | 24
 2    | 2           | 25
 3    | 3           | 26
 4    | 4           | 27
 5    | 5           | 28
 6    | 6           | 19
 7    | 7           | 22

### ABI2CH

ATmega328p ADC channel for current sensor of the secondary battery.

Type     | Size
-------- | --------
enum     | 1 byte 

Value | ADC Channel | ATMega pin
----- | ----------- | ----------
 0    | 0           | 23
 1    | 1           | 24
 2    | 2           | 25
 3    | 3           | 26
 4    | 4           | 27
 5    | 5           | 28
 6    | 6           | 19
 7    | 7           | 22

### ABU1F

Voltage factor = Umax / `ADCUREF` for flight battery.

For example, Umax = 16.8V, `ADCUREF` = 5.1V, `ABU1F` = 16.8 / 5.1 = 3.294

Type     | Size
-------- | --------
float    | 4 bytes

### ABU2F

Voltage factor = Umax / `ADCUREF` for secondary battery.

Type     | Size
-------- | --------
float    | 4 bytes

### ABI1F

Current factor = Imax / `ADCUREF` for flight batetry.

For example, Imax = 90A, `ADCUREF` = 5.1V, `ABI1F` = 90 / 5.1 = 17.647

Type     | Size
-------- | --------
float    | 4 bytes

### ABI2F

Current factor = Imax / `ADCUREF` for secondary batetry.

Type     | Size
-------- | --------
float    | 4 bytes


## ADCRSSI module settings

### ARCH

ATmega328p ADC channel for RSSI.

Type     | Size
-------- | --------
enum     | 1 byte 

Value | ADC Channel | ATMega pin
----- | ----------- | ----------
 0    | 0           | 23
 1    | 1           | 24
 2    | 2           | 25
 3    | 3           | 26
 4    | 4           | 27
 5    | 5           | 28
 6    | 6           | 19
 7    | 7           | 22

### ARINT

Interval between RSSI updates, ms. 200 is good enough.

Type     | Size
-------- | --------
uint16   | 2 bytes

### ARF

RSSI multiplier = 100 / Umax, where Umax <= `ADCUREF`.

For example, Umax = 5V, `ADCUREF` = 5.1, `ARF` = 100 / 5.0 = 20.0

Type     | Size
-------- | --------
float    | 4 bytes

## UAVTalk module settings

### UTBR

UART bitrate to communicate with FC. Typically is **3** (57600).

Type     | Size
-------- | --------
enum     | 1 byte 

Value | Bitrate
----- | ---------
 0    | 9600
 1    | 19200
 2    | 38400
 3    | 57600
 4    | 115200

### UTREL

UAVTalk version release.

**Please pay attention to this parameter! Most of the glitches or freezes caused by incorrect UTREL value.**

Type     | Size
-------- | --------
enum     | 1 byte 

Value | Firmware   | Version
----- | ---------- | ---------
 0    | OpenPilot  | 15.02.02
 1    | OpenPilot  | 15.05.02
 2    | LibrePilot | 15.09
 3    | TauLabs    | 20151123
 4    | LibrePilot | 16.09-RC1
 5    | dRonin     | 2017-02-13.1 "Artifice"
 6    | dRonin     | 2017-07-17 "Neat"

### UTIHC

Calculate home position by OSD if 1. Useful with CC3D/Atom.
If 0, flight controller defines home position.

Type     | Size
-------- | --------
bool     | 1 byte

### UTER

Emulate RSSI value based on the receiver connection state.

Type     | Size
-------- | --------
bool     | 1 byte

### UTAH

Calculate heading based on the flight controller attitude.
Set it to 0 if you want to use mag sensors ot GNSS data.

Type     | Size
-------- | --------
bool     | 1 byte

## MAVLink module settings

### MLBR

UART bitrate to communicate with FC. Typically is **57600**.

Type     | Size
-------- | --------
enum     | 1 byte 

Value | Bitrate
----- | ---------
 0    | 9600
 1    | 19200
 2    | 38400
 3    | 57600
 4    | 115200
 
### MLFC

Flight controller type. Useful with Cleanflight MAVLink telemetry: set it to **3** (APM) for
correct display of the current flight mode.

Type     | Size
-------- | --------
enum     | 1 byte 

Value | FC type
----- | -------
 0    | Autodetect
 3    | APM
 9    | Paparazzi
 12   | PX4

### MLIBL

Calculate battery level by OSD, not by flight controller if set to 1. Setting this option to 1 can
improve accuracy of the battery level calculation.

Type     | Size
-------- | --------
bool     | 1 byte

### MLBLT

Battery level warning threshold, percents. Used when `MLIBL` is 0.

Type     | Size
-------- | --------
uint8    | 1 byte

### MLER

Emulate RSSI level by receiver input channel value. Useful when receiver
has no RSSI level output.
RSSI low level warning will be set if value of the input channel defined
by `MLERC` becomes lower than `MLERT`.

Type     | Size
-------- | --------
bool     | 1 byte

### MLERC

Receiver channel for RSSI emulation. Used when `MLER` is 1.

Type     | Size
-------- | --------
enum     | 1 byte 

Value | Input channel
----- | ---------------------
 0    | Channel 1 (Roll)
 1    | Channel 2 (Pitch)
 2    | Channel 3 (Throttle)
 3    | Channel 4 (Yaw)
 4    | Channel 5
 5    | Channel 6
 6    | Channel 7
 7    | Channel 8

### MLERT

RSSI emulation threshold (minimal pulse width of the input channel), microseconds.
Used when `MLERT` = 1. Typical value is about **900**.

Type     | Size
-------- | --------
uint16   | 2 bytes


## UBX module settings

### UBXBR

UART bitrate to communicate with U-BLOX GPS module.

Type     | Size
-------- | --------
enum     | 1 byte 

Value | Bitrate
----- | ---------
 0    | 9600
 1    | 19200
 2    | 38400
 3    | 57600
 4    | 115200

### UBXTO

Connection timeout, ms. Recommended value is **1000**.

Type     | Size
-------- | --------
uint16   | 2 bytes


doc/configure.md
## Console connection

MultiOSD offers the low-level configuration console. To use it you need
terminal software and serial port (it can be any USB-UART connector, like FTDI
or Prolific pl2303).

* Run your terminal software.
* Connect MultiOSD to the serial port.
* Open the port with baudrate **57600**.
* Press reset button on MultiOSD.
* Wait for message **`READY`**.
* Type **`config`** and press `Enter`. Entered characters will not be displayed.
  If you were mistaken you'll see **`BOOT`** message. Just press reset and
  try again.

_**OR**_

* Use the tool: 

        python tools/console.py -p <port> --timeout <seconds>
   
   For example:

        python tools/console.py -p /dev/ttyUSB0 --timeout 10

* After that you'll see the command prompt

        MultiOSD v0.12
        osd#

* Type **`help`** to see commands.


## Console commands

### help

Provides information about commands.

*Example*

    osd# help
    help - Commands list
    info - Firmware version, modules, panels
    opt - Read/write OSD options
    scr - View/edit screens layout
    reset - Reset settings to defaults
    font - Upload and download mcm-file
    adc - Read ADC values
    exit - Exit console
    reboot - Reboot OSD

### info

Provides information about firmware and shows the list of available screen panels.

*Example*

    osd# info
    VERSION: 0010
    MODULES: ADCBattery ADCRSSI MAVLink
    PANELS:
    000: StableAlt
    001: Climb
    002: FlightMode
    003: ArmedFlag
    ...
    023: Compass

### exit

Exit console and run OSD.

### reboot

Reboot MinimOSD.

### opt

Read and write [options](Configuration-options).

#### opt l

List all the options with their values.

*Example*

    osd# opt l
    0x10    (byte:1)        VMODE   = 0
    0x11    (byte:1)        VBRIGHT = 0
    0x50    (str:6)         CSIGN   = -----
    ...
    0x75    (word:2)        SWCHMAX = 1700

#### opt g

Read option value from EEPROM.

*Usage:* `opt g <name>`
   - `name` - case insensitive option name

*Example*

    osd# opt g abcmul
    0x1a    (float:4)       ABCMUL  = 18.1818

#### opt s

Write option value to EEPROM.

*Usage:* `opt s <name> <value>`
   - `name` - case insensitive option name
   - `value` - option value

*Example*

    osd# opt s abcmul 19.43
    0x1a    (float:4)       ABCMUL  = 19.4300

### scr

Setup layout of the OSD screens using panels.

#### scr d

Dump current screen layout.

*Usage:* `scr d [ <screen> ]`.
   - `screen` - screen number, 0..7. All screens will be dumped when no argument supplied.

*Example*

    osd# scr d 0
    Screen 0
    #       x       y       panel   panel name
    --      --      --      --      --
    0       1       1       8       GPS
    1       8       1       9       Lat
    ...
    19      22      13      15      BatCurrent
    20      22      14      16      BatConsumed

#### scr a

Append panel to the screen layout.

*Usage:* `scr a <screen> <x> <y> <panel>`
   - `screen` - screen number, 0..7
   - `x` - X-cordinate of the panel
   - `y` - Y-cordinate of the panel
   - `panel` - panel index listed in `info` command output.

*Example*

    osd# scr d 7
    Screen 7
    #       x       y       panel   panel name
    --      --      --      --      --

    osd# scr a 7 2 3 17
    Screen 7
    #       x       y       panel   panel name
    --      --      --      --      --
    0       2       3       17      RSSIFlag

#### scr e

Edit screen panel.

*Usage:* `scr e <screen> <panel #> <x> <y> <panel>`
   - `screen` - screen number, 0..7
   - `panel #` - panel index in screen layout
   - `x` - X-cordinate of the panel
   - `y` - Y-cordinate of the panel
   - `panel` - panel index listed in `info` command output.

*Example*

    osd# scr d 2
    Screen 2
    #       x       y       panel   panel name
    --      --      --      --      --
    0       0       0       4       ConState
    1       3       0       3       ArmedFlag
    2       20      1       22      RSSI
    3       24      0       2       FlightMode
    4       1       6       7       Pitch

    osd# scr e 2 0 1 1 21
    Screen 2
    #       x       y       panel   panel name
    --      --      --      --      --
    0       1       1       21      Temperature
    1       3       0       3       ArmedFlag
    2       20      1       22      RSSI
    3       24      0       2       FlightMode
    4       1       6       7       Pitch

#### scr r

Remove panel from the screen layout.

*Usage:* `scr r <screen> <panel #>`
   - `screen` - screen number, 0..7
   - `panel #` - panel index in screen layout

*Example*

    osd# scr d 2
    Screen 2
    #       x       y       panel   panel name
    --      --      --      --      --
    0       1       1       21      Temperature
    1       3       0       3       ArmedFlag
    2       20      1       22      RSSI
    3       24      0       2       FlightMode
    4       1       6       7       Pitch

    osd# scr r 2 3
    Screen 2
    #       x       y       panel   panel name
    --      --      --      --      --
    0       1       1       21      Temperature
    1       3       0       3       ArmedFlag
    2       20      1       22      RSSI
    3       1       6       7       Pitch

#### scr c

Clear screen.

Usage: `scr c <screen>`
   - `screen` - screen number, 0..7

*Example*

    osd# scr d 2
    Screen 2
    #       x       y       panel   panel name
    --      --      --      --      --
    0       1       1       21      Temperature
    1       3       0       3       ArmedFlag
    2       20      1       22      RSSI
    3       1       6       7       Pitch

    osd# scr c 2
    Screen 2
    #       x       y       panel   panel name
    --      --      --      --      --

### font

Upload and download MAX7456 MCM-font.

#### font u

Upload MCM-file.

#### font d

Download current MAX7456 font as MCM-file.

### adc

Read ADC values.

*Usage:* `adc [ <channel> ]`
   - `channel` - ADC channel numer, 0..15

*Example*

    osd# adc
    ADC 0   0.4877 (454)
    ADC 1   0.7391 (688)
    ADC 2   0.9346 (870)
    ADC 3   1.0581 (985)
    ADC 4   1.0989 (1023)
    ADC 5   1.0989 (1023)
    ADC 6   1.0989 (1023)
    ADC 7   1.0989 (1023)
    ADC 8   0.3803 (354)
    ADC 9   0.0021 (2)
    ADC 10  0.8594 (800)
    ADC 11  0.0000 (0)
    ADC 12  0.8551 (796)
    ADC 13  0.0000 (0)
    ADC 14  0.8583 (799)
    ADC 15  0.0000 (0)


firmware/config.h
/*
 * This file is part of MultiOSD <https://github.com/UncleRus/MultiOSD>
 *
 * MultiOSD is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
#ifndef CONFIG_H_
#define CONFIG_H_

#include "defaults.h"

#define VERSION _VER(0, 19)

/*
 * Telemetry config
 */
#define TELEMETRY_UART uart0
#define TELEMETRY_CALLSIGN_LENGTH 8                       // Max. callsign chars
#define TELEMETRY_DEFAULT_BATTERY_MAX_CELL_VOLTAGE 4.2    // Maximal cell voltage, 4.2 for LiPo
#define TELEMETRY_DEFAULT_BATTERY_NOM_CELL_VOLTAGE 3.7    // Nominal cell voltage, 3.7 for LiPo
#define TELEMETRY_DEFAULT_BATTERY_MIN_CELL_VOLTAGE 3.2    // Minimal cell voltage (dead battery), 3.2 for LiPo
#define TELEMETRY_DEFAULT_BATTERY_LOW_CELL_VOLTAGE 3.5    // Warning threshold
#define TELEMETRY_DEFAULT_CALLSIGN "DRONE"                // Default callsign
#define TELEMETRY_DEFAULT_RSSI_LOW_THRESHOLD 10           // RSSI low threshold, %

/*
 * UAVTalk config
 */
#define UAVTALK_DEFAULT_BITRATE UART_BR_57600
#define UAVTALK_DEFAULT_RELEASE UAVTALK_RELEASE_DR201702131
#define UAVTALK_CONNECTION_TIMEOUT 6000                   // ms
#define UAVTALK_GCSTELEMETRYSTATS_UPDATE_INTERVAL 500     // ms
#define UAVTALK_DEFAULT_INTERNAL_HOME_CALC 1              // Home distance/direction calculation: 0 - by flight controller, 1 - by MultiOSD (GPS-based)
#define UAVTALK_DEFAULT_EMULATE_RSSI 0                    // Emulate RSSI value based on input connection state. It's always on with OpenPilot/LibrePilot
#define UAVTALK_DEFAULT_ATT_HEADING 1                     // Use attitude data for heading source

/*
 * MAVLink config
 */
#define MAVLINK_DEFAULT_BITRATE UART_BR_57600
#define MAVLINK_DEFAULT_FC_TYPE 0                         // 0 - Auto, 3 - APM, 9 - PPZ, 12 - PX4
#define MAVLINK_SYSID 'R'                                 // MAVLink system ID
#define MAVLINK_COMPID 1                                  // MAVLink component ID
#define MAVLINK_CONNECTION_TIMEOUT 2000                   // ms
#define MAVLINK_DEFAULT_INTERNAL_BATT_LEVEL 1             // Internal calculation of the battery level (may be better)
#define MAVLINK_DEFAULT_BATTERY_LOW_THRESHOD 15           // Low battery threshold, %. Only for FC battery level
#define MAVLINK_DEFAULT_RSSI_LOW_THRESHOLD 10             // RSSI low threshold, %
#define MAVLINK_DEFAULT_EMULATE_RSSI 0                    // Emulate RSSI by input channels
#define MAVLINK_DEFAULT_EMULATE_RSSI_CHANNEL 2            // Input channel number for RSSI emulation 0 - roll, 1 - pitch, 2 - throttle, ...
#define MAVLINK_DEFAULT_EMULATE_RSSI_THRESHOLD 920        // Input channel threshod for RSSI emulation

/*
 * UBX config
 */
#define UBX_DEFAULT_BITRATE UART_BR_19200                 // values higher than 57600 is not recommended
#define UBX_DEFAULT_TIMEOUT 1000                          // ms
#define UBX_DEFAULT_AUTOCONF 0                            // automatically configure GPS-module after connect

/*
 * MSP config
 */
#define MSP_DEFAULT_BITRATE UART_BR_115200
#define MSP_CONNECTION_TIMEOUT 2000                       // ms
#define MSP_DEFAULT_INTERNAL_HOME_CALC 0                  // Home distance/direction calculation: 0 - flight controller, 1 - MultiOSD by GPS
#define MSP_DEFAULT_AMPERAGE_DIVIDER 10                   // MSP_ANALOG amperage factor (100 or 1000)
#define MSP_DEFAULT_GPS_ALTITUDE 0                        // Use GPS altitude instead of computed by FC

/*
 * OSD config
 */
#define OSD_SCREEN_PANELS 24                              // (OSD_SCREEN_PANELS * 3) bytes in SRAM
#define OSD_MAX_SCREENS 8

#define OSD_DEFAULT_SCREENS 8                             // each screen will consume (OSD_SCREEN_PANELS * 3) bytes in EEPROM
#define OSD_DEFAULT_SWITCH_MODE OSD_SWITCH_MODE_TOGGLE    // OSD_SWITCH_MODE_SELECTOR or OSD_SWITCH_MODE_TOGGLE
#define OSD_DEFAULT_SWITCH_CHANNEL 6                      // switch input channel
#define OSD_DEFAULT_SELECTOR_MIN 1000                     // us, minimal input channel value for selector mode
#define OSD_DEFAULT_SELECTOR_MAX 2000                     // us, maximal input channel value for selector mode
#define OSD_DEFAULT_TOGGLE_TRESHOLD 1200                  // us, input channel value threshold for toggle mode

/*
 * UART config
 */
#define UART_STDIO                                        // we need fprintf
#define UART_RX_BUFFER_SIZE 256                           // I like big buffers
#define UART_TX_BUFFER_SIZE 64

/*
 * Console config
 */
#define CONSOLE_UART uart0
#define CONSOLE_MAX_CMD_LENGTH 32
#define CONSOLE_EOL "\r\n"
#define CONSOLE_PROMPT "osd# "
#define CONSOLE_BITRATE UART_BR_57600

/*
 * Boot config
 */
#define BOOT_CONFIG_CODE "config"
#define BOOT_CONFIG_WAIT_TIME 8000

/*
 * ADC config
 */
#define ADC_DEFAULT_REF ADC_REF_INTERNAL                  // ADC reference voltage source: ADC_REF_AREF (0) / ADC_REF_AVCC (1) / ADC_REF_INTERNAL (2)
#define ADC_DEFAULT_REF_VOLTAGE 1.1                       // ADC reference voltage value

/*
 * ADC battery
 */
#define ADC_BATTERY_DEFAULT_UPDATE_INTERVAL 200           // ms
#define ADC_BATTERY_DEFAULT_VOLTAGE1_CHANNEL 2            // ADC2 25 pin
#define ADC_BATTERY_DEFAULT_VOLTAGE2_CHANNEL 0            // ADC0 23 pin
#define ADC_BATTERY_DEFAULT_AMPERAGE1_CHANNEL 1           // ADC1 24 pin
#define ADC_BATTERY_DEFAULT_AMPERAGE2_CHANNEL 15          // 0V
#define ADC_BATTERY_DEFAULT_VOLTAGE1_FACTOR 1.0
#define ADC_BATTERY_DEFAULT_VOLTAGE2_FACTOR 1.0
#define ADC_BATTERY_DEFAULT_AMPERAGE1_FACTOR 1.0
#define ADC_BATTERY_DEFAULT_AMPERAGE2_FACTOR 1.0

/*
 * ADC RSSI
 */
#define ADC_RSSI_DEFAULT_CHANNEL 3                        // ADC3 26 pin
#define ADC_RSSI_DEFAULT_UPDATE_INTERVAL 200              // ms
#define ADC_RSSI_DEFAULT_MULTIPLIER 1.0


/*
 * SPI config
 */
#define SPI_PORT PORTB
#define SPI_DDR DDRB
#define SPI_SS_BIT PB2
#define SPI_MOSI_BIT PB3
#define SPI_MISO_BIT PB4
#define SPI_SCK_BIT PB5
#define SPI_CPOL 0
#define SPI_CPHA 0
#define SPI_FREQ 8000000L

/*
 * MAX7456 config
 */
#define MAX7456_DEFAULT_MODE MAX7456_MODE_PAL             // default video mode, if jumper closed
#define MAX7456_DEFAULT_BRIGHTNESS 0x00                   // 120% white, 0% black

// ~CS (Chip Select) pin
#define MAX7456_SELECT_PORT PORTD
#define MAX7456_SELECT_DDR DDRD
#define MAX7456_SELECT_BIT PD6

// VSYNC port
#define MAX7456_VSYNC_PORT PORTD
#define MAX7456_VSYNC_PIN PIND
#define MAX7456_VSYNC_DDR DDRD
#define MAX7456_VSYNC_BIT PD2

// Video mode pin. High = default, low = autodetect
#define MAX7456_MODE_JUMPER_PORT PORTD
#define MAX7456_MODE_JUMPER_DDR DDRD
#define MAX7456_MODE_JUMPER_BIT PD3


#endif /* CONFIG_H_ */
